### PR TITLE
Create helper functions to handle WebDriver actions. Some actions on base …

### PR DESCRIPTION
**Pull request by angjieli**
…FlutterDriver will now throw unimplemented error.

## Description

- Create some helper functions to handle WebDriver actions. 
- Some actions on base FlutterDriver will now throw unimplemented error.]

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_driver/lib/src/driver/web_driver.dart [hunk 1]
```diff
@@ -248,3 +227,30 @@ class FlutterWebConnection {
     _driver.quit();
   }
 }
+
+/// Utility functions for working with WebDriver.
+class DriverUtils {
```
Comment (jonahwilliams) [inline packages/flutter_driver/lib/src/driver/web_driver.dart]: I don't think there isn't any reason to make a class with only static members in Dart. Prefer creating top-level functions (see https://dart.dev/guides/language/effective-dart/design#avoid-defining-a-class-that-contains-only-static-members )

File: packages/flutter_driver/lib/src/driver/web_driver.dart [hunk 2]
```diff
@@ -248,3 +227,30 @@ class FlutterWebConnection {
...
+    }
+  }
+
+  /// Waits until Extension is installed.
+  static Future<void> waitUntilExtensionInstalled(sync_io.WebDriver driver, Duration timeout) async {
+    await waitFor<void>(() => driver.execute('return typeof(window.\$flutterDriver)', <String>[]),
+        matcher: 'function',
+        timeout: timeout ?? const Duration(days: 365));
+  }
+}
```
Comment (jonahwilliams) [inline packages/flutter_driver/lib/src/driver/web_driver.dart]:
```suggestion
}

```

File: packages/flutter_driver/lib/src/driver/driver.dart [hunk 3]
```diff
@@ -157,10 +157,10 @@ abstract class FlutterDriver {
   }
 
   /// Getter of appIsolate
-  VMIsolate get appIsolate;
+  VMIsolate get appIsolate => throw UnimplementedError();
```
Comment (jonahwilliams) [inline packages/flutter_driver/lib/src/driver/driver.dart]: What is the rationale for throwing versus abstract? Just curious...

File: packages/flutter_driver/lib/src/driver/web_driver.dart (hunk 1 above)
Comment (angjieli) [reply packages/flutter_driver/lib/src/driver/web_driver.dart]: Done.

File: packages/flutter_driver/lib/src/driver/driver.dart (hunk 3 above)
Comment (angjieli) [reply packages/flutter_driver/lib/src/driver/driver.dart]: Discussed with Jonah offline.

**Review by jonahwilliams — APPROVED** [review]
LGTM

The code changes are refactors of existing functionality

**Pull request merged**